app/services/articles/feed.rb

```diff
@@ -115,11 +115,40 @@ def mix_of_everything_experiment
         more_comments_experiment
       when 6
         more_experience_level_weight_experiment
+      when 7
+        more_tag_weight_randomized_at_end_experiment
+      when 8
+        more_experience_level_weight_randomized_at_end_experiment
+      when 9
+        more_comments_randomized_at_end_experiment
       else
         default_home_feed(user_signed_in: true)
       end
     end
 
+    # Randomized at end group for next three experiments
+    # Rather than randomizing *during* the ranking, rank solely by quality/match
+    # and randomize the top half of the ranked results.
+    # Resulting in more relevance and still more freshness.
+
+    def more_tag_weight_randomized_at_end_experiment
+      @randomness = 0
+      results = more_tag_weight_experiment
+      first_half(results).shuffle + last_half(results)
+    end
+
+    def more_experience_level_weight_randomized_at_end_experiment
+      @randomness = 0
+      results = more_experience_level_weight_experiment.shuffle
```

Comment:
Should this `shuffle` be here and in the method below? Given the PR description I was under the impression we just wanted to shuffle the first half.

Reply:
Oh yeah. That's a mistake. Nice catch.